src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/model/ValidatableActionResult.kt

```diff
@@ -0,0 +1,7 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.model
+
+sealed interface ValidatableActionResult<ResultType> {
+  data class Success<ResultType>(val result: ResultType) : ValidatableActionResult<ResultType>
```

Comment:
I think `entity` and `EntityType` would be better names here, i.e:

```suggestion
  data class Success<EntityType>(val entity: EntityType) : ValidatableActionResult<EntityType>
```